Applications/IRCClient/IRCClient.cpp

```diff
@@ -687,32 +687,75 @@ void IRCClient::handle_user_command(const String& input)
         return;
     }
     if (command == "/PART") {
-        if (parts.size() >= 2)
-            part_channel(parts[1]);
+        if (parts.size() >= 2) {
+            auto channel = parts[1];
+            part_channel(channel);
+        } else {
+            auto* window = current_window();
+            if (!window || window->type() != IRCWindow::Type::Channel)
+                return;
+            auto channel = window->channel().name().characters();
+            join_channel(channel);
+        }
         return;
     }
     if (command == "/HOP") {
         if (parts.size() >= 2) {
-            part_channel(parts[1]);
-            join_channel(parts[1]);
+            auto channel = parts[1];
+            part_channel(channel);
+            join_channel(channel);
+	} else {
+            auto* window = current_window();
+            if (!window || window->type() != IRCWindow::Type::Channel)
+                return;
+            auto channel = window->channel().name().characters();
```

Comment:
Let's avoid using raw pointers like this if we can avoid it.
```suggestion
            auto channel = window->channel().name();
```